### ISSUE TITLE
DEVELOPER.md up to date?

### ISSUE THREAD

**Issue by joshbabb** [driftyco/ionic-native#1210]
Hi, I'm going through the process of testing and finalizing a feature branch for ionic-native and I think that DEVELOPER.md is out of date--shouldn't we be using @ionic-native? :)

**Comment by joshbabb** [comment]
proposed fix in https://github.com/joshbabb/ionic-native/commit/fd3b8d0ea4f216433652e715e6fb06b2e88317ee

**Comment by ihadeed** [comment]
Yes. Sorry, we haven't updated `DEVELOPER.md` for Ionic Native 3.x yet.

**Comment by ihadeed** [comment]
Just had a look at the commit you mentioned. The new text is partially correct.

According to the current process:
- you need to copy `/dist/packages-dist/@ionic-native/core/` to `YOUR_APP/node_modules/@ionic-native/core/`
- then copy any plugins that you are using from `/dist/packages-dist/@ionic-native/plugins/` to `YOUR_APP/node_modules/@ionic-native/`

I think this need to be changed though. It would make more sense to get rid of both `packages-dist` and `plugins` folders, so the `dist/@ionic-native/` directory will match the `/node_modules/@ionic-native/` directory.

**Comment by joshbabb** [comment]
Then we need to update our DEVELOPER.md in https://github.com/driftyco/ionic-native/pull/1212

Do note, I have already tested that our custom ionic-native fork supporting the JINS MEME smart glasses works according to the steps in the current text.  Looking at this...

**Comment by joshbabb** [comment]
Alright, sorry about that, but all this should be tested working and the docs have been updated as of https://github.com/BlyncSync/ionic-native/commit/484e7a14c3892adc71fadc4770cd00ed653e29fa

**Comment by ehorodyski** [comment]
How would you test a newly added plugin since each plugin is it's own dependency?

**Comment by joshbabb** [comment]
Eric,

How we did it was to fork ionic-native, add our own custom plugin code,
`npm run build` our fork, and manually copy over the @ionic-native folder
produced into the node_modules folder of the test project (or use a
relative path instead of a version number in package.json)

Sorry if that's not what you were asking

**Comment by ehorodyski** [comment]
@joshbabb -- I did that as well. 

The Ionic Native 3.x structure has each plugin as it's own node dependency, so this is what my `package.json` is looking like:

```json
"dependencies": {
    "@angular/common": "2.4.8",
      ...
    "@ionic-native/core": "3.1.0",
    "@ionic-native/splash-screen": "3.1.0",
    "@ionic-native/status-bar": "3.1.0",
    "@ionic-native/intel-security": "file:node_modules/ionic-native/dist/@ionic-native/intel-security",
     ...
    "zone.js": "0.7.2"
  },
```

But when I a) try `npm install @ionic-native/intel-security` or b) try to just reference it in the code, I get:

```shell
npm ERR! git C:\Users\ehorodys\Desktop\test\node_modules\ionic-native: Appears to be a git repo or submodule.
```

I did copy the entire `ionic-native` project into `node_modules/ionic-native` however, maybe I need to remove anything that isn't `dist`?

**Comment by joshbabb** [comment]
I did not use the file: method of linking the correct package, I used ../
(relative paths.)

Also make sure that you're using the new @ionic-native

Will post more once I'm at my desk.

**Comment by ehorodyski** [comment]
Hey @joshbabb -- Thanks for the response.

I got it working, I moved over the entire `ionic-native` package, and it literally just needs to be `dist`. I should have listened to the docs, but I wasn't sure if it was accurate. I thought that since each plugin is it's own module that it would follow a folder structure in `node_modules`  where each plugin got it's own folder.

Very happy it isn't, but that was my line of thinking.

Thanks again for the help!
- Eric
